pocs/camera/camera.py

```diff
@@ -275,23 +280,36 @@ def take_exposure(self,
         if not isinstance(seconds, u.Quantity):
             seconds = seconds * u.second
 
-        self.logger.debug('Taking {} second exposure on {}: {}'.format(
+        self.logger.debug('Taking {} exposure on {}: {}'.format(
             seconds, self.name, filename))
 
-        exposure_event = threading.Event()
         header = self._fits_header(seconds, dark)
 
-        self._take_exposure(seconds=seconds,
-                            filename=filename,
-                            dark=dark,
-                            exposure_event=exposure_event,
-                            header=header,
-                            *args, **kwargs)
+        if not self._exposure_event.is_set():
+            self.logger.warning('Exposure started on {} while one in progress! Waiting.'.format(
+                self))
+            self._exposure_event.wait()
+
+        # Clear event now to prevent any other exposures starting before this one is finsihed.
+        self._exposure_event.clear()
+
+        try:
+            # Camera type specific exposure set up and start
+            readout_args = self._start_exposure(seconds, filename, dark, header, *args, *kwargs)
```

Comment:
Could stash them in a private member variable during `start_exposure` I suppose instead of  returning them and passing them through `_poll_exposure` to `_readout`. Not sure that'd be an improvement though as it kind of hides the fact that `start_exposure` needs to pass information to `_readout`.